src/validation.cpp

```diff
@@ -3407,7 +3408,11 @@ bool Chainstate::ActivateBestChain(BlockValidationState& state, std::shared_ptr<
                     break;
                 }
             }
-        }
+            } // release MempoolMutex
+            if (m_chainman.m_options.signals) {
+                m_chainman.m_options.signals->UpdatedBlockTipSync(pindexNewTip);
```

Comment:
This call invokes logging, i.e. I/O operations, while holding the `::cs_main` and `m_mempool->cs` mutexes. It is considered as a suboptimal approach from the performance point of view. Could it be avoided?

Reply:
I don't think it's true that `m_mempool->cs` is still held here (see line above)?

Reply:
As for `cs_main`, it seems necessary for the chain tip to be what we think it is when the callback happens. IIUC the logging only happens in debug?